runtime/mainnet/src/weights/xcm/mod.rs

```diff
@@ -0,0 +1,312 @@
+mod pallet_xcm_benchmarks_fungible;
+mod pallet_xcm_benchmarks_generic;
+
+use alloc::vec::Vec;
+
+use frame_support::BoundedVec;
+use pallet_xcm_benchmarks_fungible::WeightInfo as XcmFungibleWeight;
+use pallet_xcm_benchmarks_generic::WeightInfo as XcmGeneric;
+use xcm::{
+	latest::{prelude::*, AssetTransferFilter},
+	DoubleEncoded,
+};
+
+use crate::{config::xcm::MaxAssetsIntoHolding, Runtime};
+
+trait WeighAssets {
+	fn weigh_assets(&self, weight: Weight) -> Weight;
+}
+
+const MAX_ASSETS: u64 = 100;
+
+impl WeighAssets for AssetFilter {
+	fn weigh_assets(&self, weight: Weight) -> Weight {
+		match self {
+			Self::Definite(assets) => weight.saturating_mul(assets.inner().iter().count() as u64),
+			Self::Wild(asset) => match asset {
+				All => weight.saturating_mul(MAX_ASSETS),
+				AllOf { fun, .. } => match fun {
+					WildFungibility::Fungible => weight,
+					// Magic number 2 has to do with the fact that we could have up to 2 times
+					// MaxAssetsIntoHolding in the worst-case scenario.
```

Comment:
Reason for this is:

> // worst-case, holding.len becomes 2 * holding_limit.
> // this guarantees that if holding.len() == holding_limit and you have more than
> // `holding_limit` items (which has a best case outcome of holding.len() == holding_limit),
> // then the operation is guaranteed to succeed.

[source](https://github.com/paritytech/polkadot-sdk/blob/4a400dc1866f11707331fb6408df1055d0f42a70/polkadot/xcm/xcm-executor/src/lib.rs#L470-L473)

Also, even though I've found this comment: https://github.com/paritytech/cumulus/pull/2102#discussion_r1100706750
The reason why this only is applied for `NonFungible` is not that apparent to me. It seems that it would apply to both cases.

Reply:
Worth pinging a Cisco or someone about?

Reply:
Clarified in: [0a50386](https://github.com/r0gue-io/pop-node/pull/479/commits/0a5038643b8521466f79fd261800133a00800c01)